### PR TITLE
Handle Set-Cookie Response Header

### PR DESCRIPTION
**Pull request by MikeZeDev**
Fixes https://github.com/manga-download/haruneko/issues/98 (website still bugged)

Its a first attempt there is certainly an issue with chrome.cookies.set.

### CODE REVIEW

File: web/src/engine/FetchProviderNodeWebkit.ts [hunk 1]
```diff
@@ -46,6 +46,30 @@ function ModifyRequestHeaders(details: chrome.webRequest.WebRequestHeadersDetail
 }
 
 function ModifyResponseHeaders(details: chrome.webRequest.WebResponseHeadersDetails): chrome.webRequest.BlockingResponse {
+
+    //Handle Set-Cookie header according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+    const cookiesHeaders = details.responseHeaders.filter(header => header.name.toLowerCase() == 'set-cookie');
+
+    //Multiple Set-Cookie headers can be sent in the same response.
+    if (cookiesHeaders.length > 0) {
```
Comment (Sheepux) [inline web/src/engine/FetchProviderNodeWebkit.ts:54]: no need for this `if` as  `foreach` will iterate only on elements.
filter can be `null` then just use `cookiesHeaders?.forEach`

**Comment by MikeZeDev** [comment]
Testing this for a while, go errors 
image

Maybe we shouldnt remove space, nor use lowercase. I feel like it will be safer to compare the key name to an exhaustive of attribute (path, expires, httponly, etc...) and fill details using the matching key rather than directly using the key from the SetCookie header.

**Comment by MikeZeDev** [comment]
`FetchProviderNodeWebkit.ts:80 Uncaught (in promise) TypeError: Error in invocation of cookies.set(object details, optional function callback): Error at parameter 'details': Error at property 'expirationDate': Invalid type: expected number, found string.`

Damn, i know why but seriously...

**Comment by ronny1982** [comment]
Use `FetchBrowser` if you want cookies from the web server to be set in HakuNeko.
Self-managing cookies should be avoided.

**Comment by MikeZeDev** [comment]
> Use `FetchBrowser` if you want cookies from the web server to be set in HakuNeko. Parsing and self-managing response cookies should be avoided.

FetchBrowser ? I see no FetchBrowser in code.

> May also try to use the Fetch API with option credentials: same-origin/include, maybe this will allow fetch to set cookies automatically

Cookies to be sent arent in the request because they are not set at all on browser storage in the first place. Using includes indeed send everything but there is not the new cookie stored

1) I use FetchRequest(chapterURL)
2) Response is code 302. Request move to other location
3) Following response  302 again and this time there is a Setcookie header  with "JSESSIONID=xxxxxxxx"
4) Last location request doesnt get the cookie from SetHeader according to console because its even missing from cookie storage. Page displayed is garbage .

Meanwhile on Firefox : 
1) Request (chapterURL)
2) Response is code 302. Request move to other location
3) Following response  302 again and this time there is a Setcookie header  with "JSESSIONID=xxxxxxxx"
4) Last location request get the cookie from SetHeader. Page displayed is chapter viewer

Related :  https://stackoverflow.com/questions/61555100/cookie-in-set-cookie-header-not-being-set
https://stackoverflow.com/questions/31058542/set-cookie-header-not-setting-cookie-in-chrome

**Comment by ronny1982** [comment]
Suggestion to abandon this PR for the following reasons:
- The `Fetch API` is usually not supposed to process the `Set-Cookie` headers and HakuNeko should comply with browser behavior as much as possible (otherwise it might be used by websites to detect and block HakuNeko e.g., `'Set-Cookie: HakuNeko=TRUE'`)
- Maintaining and testing the cookie parser to work correctly with all possible permutations of `Set-Cookie` header values is additional work which should be avoided
- The `FetchProvider` module already has methods which will process the `Set-Cookie` headers
  - `FetchWindowCSS` => now fixed
  - `FetchWindowScript`
  - `FetchWindowPreloadScript`)

**Pull request closed without merging**